File: packages/react-native/android/src/main/java/com/breezsdkliquid/BreezSDKLiquidLogStream.kt
package com.breezsdkliquid

import breez_sdk_liquid.LogEntry
import breez_sdk_liquid.Logger
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter

class BreezSDKLiquidLogger(
    private val emitter: RCTDeviceEventEmitter,
) : Logger {
    companion object {
        var emitterName = "breezSdkLiquidLog"
    }

    override fun log(l: LogEntry) {
        emitter.emit(emitterName, readableMapOf(l))
    }
}

File: packages/react-native/ios/BreezSDKLiquidLogStream.swift
import Foundation
import BreezSDKLiquid

class BreezSDKLiquidLogger: BreezSDKLiquid.Logger {
    static let emitterName: String = "breezSdkLiquidLog"

    func log(l: LogEntry) {
        if RNBreezSDKLiquid.hasListeners {
            RNBreezSDKLiquid.emitter.sendEvent(withName: BreezSDKLiquidLogger.emitterName,
                                         body: BreezSDKLiquidMapper.dictionaryOf(logEntry: l))
        }
    }
}